frontend/src/components/MeetingRoom/CaptionsButton/CaptionsButton.tsx

```diff
@@ -0,0 +1,141 @@
+import useRoomName from '../../../hooks/useRoomName';
+import ToolbarButton from '../ToolbarButton';
+import { enableCaptions } from '../../../api/captions';
+import { SubscriberWrapper } from '../../../types/session';
+import CaptionsBox from './CaptionsBox';
+import useSessionContext from '../../../hooks/useSessionContext';
+
+export type CaptionsButtonProps = {
+  isOverflowButton?: boolean;
+  handleClick?: () => void;
+  subscriberWrappers: SubscriberWrapper[];
+};
+
+/**
+ * CaptionsButton Component
+ *
+ * Displays a button and handles the captioning functionality.
+ * @param {CaptionsButtonProps} props - the props for the component
+ *  @property {boolean} isOverflowButton - (optional) whether the button is in the ToolbarOverflowMenu
+ *  @property {(event?: MouseEvent | TouchEvent) => void} handleClick - (optional) click handler that closes the overflow menu in small viewports.
+ * @returns {ReactElement} - The CaptionsButton component.
+ */
+const CaptionsButton = ({
+  isOverflowButton = false,
+  handleClick,
+  subscriberWrappers,
+}: CaptionsButtonProps): ReactElement => {
+  const { currentCaptionsIdRef, ownCaptions, vonageVideoClient } = useSessionContext();
+  const roomName = useRoomName();
+  const [captionsId, setCaptionsId] = useState<string>('');
+  const [isCaptionsEnabled, setIsCaptionsEnabled] = useState<boolean>(false);
+  const title = isCaptionsEnabled ? 'Disable captions' : 'Enable captions';
+
+  const handleClose = () => {
+    // If the CaptionsButton is in the ToolbarOverflowMenu, we close the modal and the menu
+    if (isOverflowButton && handleClick) {
+      // Close the menu immediately to improve mobile UX
+      handleClick();
+    }
+  };
+
+  const handleCaptions = async (action: 'enable' | 'disable') => {
+    if (action === 'enable') {
+      if (currentCaptionsIdRef?.current) {
+        setIsCaptionsEnabled(true);
+        setCaptionsId(currentCaptionsIdRef.current);
+
+        // we need to signal to other users that we are joining the captions
+        // This is to ensure that if the captions are already enabled, we just join them
+        // and not create a new captions session
+        vonageVideoClient?.signal({
+          type: 'captions',
+          data: JSON.stringify({
+            action: 'join',
+            captionsId: currentCaptionsIdRef.current,
+          }),
+        });
+      } else if (!captionsId && roomName) {
+        try {
+          const response = await enableCaptions(roomName);
+          setCaptionsId(response.data.captions.captionsId);
+          currentCaptionsIdRef.current = response.data.captions.captionsId;
+          setIsCaptionsEnabled(true);
+
+          // we need to signal to other users that captions have been enabled
+          vonageVideoClient?.signal({
+            type: 'captions',
+            data: JSON.stringify({
+              action: 'enable',
+              captionsId: response.data.captions.captionsId,
+            }),
+          });
```

Comment:
Could we handle this through the backend?

Reply:
we are eventually sending the request to the backend to enable the captions. I'd prefer to keep the captions management on the frontend since this way we can easily make UI changes based on the state and do not need to make additional backend requests. it's super easy to synchronize the captions state across all the users. To be fair, it'd be probably be easiest to have some sort of a database to put this all in but we don't have that so I think sticking to managing it on the frontend is a way to go